isis/tests/FunctionalTestsCtxcal.cpp

```diff
@@ -0,0 +1,125 @@
+#include "Fixtures.h"
+#include "Pvl.h"
+#include "PvlGroup.h"
+#include "TestUtilities.h"
+#include "Histogram.h"
+
+#include "ctxcal.h"
+
+#include "gtest/gtest.h"
+
+using namespace Isis;
+
+static QString APP_XML = FileName("$ISISROOT/bin/xml/ctxcal.xml").expanded();
+
+
+TEST_F(MroCtxCube, FunctionalTestCtxcalDefault) {
+  QString outCubeFileName = tempDir.path() + "/outTemp.cub";
+  QVector<QString> args = {"to="+outCubeFileName};
+
+  UserInterface options(APP_XML, args);
+  
+  try {
+    ctxcal(testCube.get(), options);
+  }
+  catch (IException &e) {
+    FAIL() << "Unable to open image: " << e.what() << std::endl;
+  }
+
+  Cube oCube(outCubeFileName, "r");
+
+  Histogram *oCubeStats = oCube.histogram();
+
+  EXPECT_DOUBLE_EQ(oCubeStats->Average(), 0.077640061192214491);
+  EXPECT_DOUBLE_EQ(oCubeStats->Sum(), 31.056024476885796);
+  EXPECT_DOUBLE_EQ(oCubeStats->ValidPixels(), 400);
+  EXPECT_DOUBLE_EQ(oCubeStats->StandardDeviation(), 0.0012347471238170408);
+}
+
+
+TEST_F(MroCtxCube, FunctionalTestCtxcalFlatfile) {
+  QString outCubeFileName = tempDir.path() + "/outTemp.cub";
+  QVector<QString> args = {"to="+outCubeFileName, "flatfile=$ISISDATA/mro/calibration/ctxFlat_0001.cub"};
+
+  UserInterface options(APP_XML, args);
+  
+  try {
+    ctxcal(testCube.get(), options);
+  }
+  catch (IException &e) {
+    FAIL() << "Unable to open image: " << e.what() << std::endl;
+  }
+
+  Cube oCube(outCubeFileName, "r");
+
+  Histogram *oCubeStats = oCube.histogram();
+
+  EXPECT_DOUBLE_EQ(oCubeStats->Average(), 0.10046864503994585);
+  EXPECT_DOUBLE_EQ(oCubeStats->Sum(), 40.187458015978336);
+  EXPECT_DOUBLE_EQ(oCubeStats->ValidPixels(), 400);
+  EXPECT_DOUBLE_EQ(oCubeStats->StandardDeviation(), 0.0018248585597074806);
```

Comment:
This method of testing looks good to me for the DN values. For calibration programs is it important to test the information that ends up in the PVL label under the `Raidometric Calibration` group? A lot of the old tests do check the labels, but I'm not sure how important that is.

Reply:
I am basing this on the fact that by checking DNs, the DNs should also change if the values under the Radiometric group also change. Although, I am not against just checking them anyways since its not really a hard check to do.